integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/samples/JSONPayloadSampleTestCase.java

```diff
@@ -109,13 +114,47 @@ public void performJsonBatchRequest() {
                 "]\n" +
                 "}\n" +
                 "}";
-        getHttpResponse(serviceEndPoint + "employee_batch_req", "POST", postInsertPaymentPayload);
+        getHttpResponse(serviceEndPoint + "employee_batch_req", "POST", postInsertPayload);
         String response_employee1 = getHttpResponse(serviceEndPoint + "employee/53", "GET", null);
         String response_employee2 = getHttpResponse(serviceEndPoint + "employee/101", "GET", null);
-        Assert.assertTrue(response_employee1.contains("{\"employees\":{\"employee\":[{\"lastName\":\"Sangeeth\",\"firstName\":\"Karunaratne\",\"salary\":18400.0}]}}"),
-                "Batch request POST method failed");
-        Assert.assertTrue(response_employee2.contains("{\"employees\":{\"employee\":[{\"lastName\":\"Smitth\",\"firstName\":\"Will\",\"salary\":15500.0}]}}"),
-                "Batch request POST method failed");
+        Assert.assertTrue(response_employee1.contains("{\"employees\":{\"employee\":[{\"lastName\":\"Sangeeth\"," +
+                "\"firstName\":\"Karunaratne\",\"salary\":18400.0}]}}"),"Batch request POST method failed");
+        Assert.assertTrue(response_employee2.contains("{\"employees\":{\"employee\":[{\"lastName\":\"Smitth\"," +
+                "\"firstName\":\"Will\",\"salary\":15500.0}]}}"),"Batch request POST method failed");
+    }
+
+    @Test(groups = "wso2.dss", description = "Invoking PUT Request without optional fields in JSON payload")
+    public void performJsonPutRequestWithoutOptionaParameter() {
+
+        String payload = "{\n" +
+                "\"employee\":{\n" +
+                "\"employeeNumber\":52,\n" +
+                "\"salary\":21400.00\n" +
+                "}\n" +
+                "}";
+        getHttpResponse(serviceEndPoint + "employee", "PUT", payload);
+        String response = getHttpResponse(serviceEndPoint + "employee/52", "GET", null);
+        Assert.assertTrue(response.contains(
+                "{\"employees\":{\"employee\":[{\"lastName\":\"Karunaratne\",\"firstName\":\"Sangeeth\"," +
+                        "\"salary\":21400.0}]}}"), "POST method failed");
+    }
+
+    @Test(groups = "wso2.dss", description = "Invoking PUT Request with optional fields in JSON payload as " +
+            "declared in the update query")
+    public void performJsonPutRequestWithOptionalParameter() {
+
+        String payload = "{\n" +
+                "\"employee\":{\n" +
+                "\"employeeNumber\":52,\n" +
+                "\"firstName\":\"Sangeeth\",\n" +
+                "\"email\":\"sangeeth@wso2.com\",\n" +
+                "\"salary\":18400.00\n" +
+                "}\n" +
+                "}";
+        getHttpResponse(serviceEndPoint + "employee", "PUT", payload);
+        String response = getHttpResponse(serviceEndPoint + "employee/52", "GET", null);
+        Assert.assertTrue(response.contains("{\"employees\":{\"employee\":[{\"lastName\":\"Karunaratne\"," +
```

Comment:
Shall we add a failure message as well ?

Reply:
Fixed